### DISCUSSION TITLE
Subproject X did not override X dependency and no variable name specified

### DISCUSSION THREAD

**Discussion by kraftwerk28** [mesonbuild/meson#10484] [Q&A] [answered]
I'm trying to build https://gitlab.gnome.org/raggesilver/blackbox, using the following `subprojects/marble.wrap` file:
```ini
[wrap-git]
url = https://gitlab.com/raggesilver/marble.git
revision = master
```
But running `meson setup --buildtype=release build` outputs:
```sh
WARNING: Subproject 'marble' did not override 'marble' dependency and no variable name specified
Dependency marble from subproject subprojects/marble found: NO
```
How this can be solved? Thank you in advance.

**Comment by eli-schwartz** [comment, marked as answer]
https://gitlab.com/raggesilver/marble/-/blob/master/src/meson.build#L51

marble declares a variable `marble_lib_dep`, and Meson is suggesting that you should specify that as documented in https://mesonbuild.com/Wrap-dependency-system-manual.html#provide-section
